### DISCUSSION TITLE
X-Cache-Status in Proxy-Cache plugin always showing Bypass

### DISCUSSION THREAD

**Discussion by tanmoyrsc** [Kong/kong#7217] [Help] [answered]
Hi,

I have used this plugin with Kong (2.0.4) Community edition but X-Cache-Status is always Bypass. have set this proxy-caching using below command for all consumers. Please help me to sort this out.

curl -X POST http://xxx.xxx.xxx.xxx:8001/services/test-proxy-cache/plugins
--data "name=proxy-cache"
--data "config.cache_ttl=300"
--data "config.strategy=memory"
--data "config.content_type=text/plain; charset=utf-8"

curl -X POST http://xxx.xxx.xxx.xxx:8001/routes/test-proxy-cache-route/plugins
--data "name=proxy-cache"
--data "config.cache_ttl=300"
--data "config.strategy=memory"
--data "config.content_type=text/plain; charset=utf-8"

Br,
Tanmoy

**Comment by khangt1k25** [comment]
I have problem with proxy-cached too, The config is right and every go right despite of bypass status. It's always show the bypass status and still make request to upstream 

< X-Cache-Key: 365a5370d8d8e456155a3758ce37517a88768017af50b5d0c76bc1adfbd27015
< X-Cache-Status: Bypass

**Reply by khangt1k25** [reply]
The upstream header response is here and i also try to on/off the cache_control config of proxy-cache plugin. It doesn't work
```
> GET /test?text=123 HTTP/1.1
> Host: 13.213.46.69:19999
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
< Content-Type: text/html; charset=utf-8
< Content-Length: 12
< length_token: 3
< Server: Werkzeug/2.0.2 Python/3.9.7
< Date: Fri, 14 Jul 2023 09:18:20 GMT

```

**Comment by raselSoftNode** [comment]
is anybody got the solution ? if yes, please share your solution with me.

**Comment by ms2008** [comment, marked as answer]
Can you try the latest version? I was not able to reproduce the problem with 3.3.0 with the following steps:

```
$ curl -i -X  POST http://localhost:8001/services/ \
  --data "name=plugin-testing" \
  --data "url=http://192.168.3.10:4000/plain"

$ curl -i -X  POST http://localhost:8001/services/plugin-testing/routes \
  --data "paths[]=/plain"

$ curl -X POST http://localhost:8001/plugins \
    --data "name=proxy-cache"  \
    --data "config.response_code=200"  \
    --data "config.request_method=GET"  \
    --data "config.request_method=HEAD"  \
    --data "config.content_type=text/html; charset=utf-8"  \
    --data "config.content_type=application/json; charset=utf-8"  \
    --data "config.cache_ttl=300"  \
    --data "config.strategy=memory"
```

After running curl multiple times, you can see that the `X-Cache-Status` response header is returned with `Hit`

```
$ curl -is http://localhost:8000/plain
HTTP/1.1 200 OK
Content-Type: text/html; charset=utf-8
Connection: keep-alive
X-Cache-Key: aa16a1dc1f68606237c11335052986e329be06b8d412e24458d3b359bacc8c58
Age: 6
X-Powered-By: Express
ETag: W/"14-YmTwcXKe7Bxv+fq55MlkxtnsVhA"
X-Cache-Status: Hit
Date: Tue, 01 Aug 2023 02:45:46 GMT
Cache-Control: max-age=300
Content-Length: 20
X-Kong-Upstream-Latency: 0
X-Kong-Proxy-Latency: 0
Via: kong/3.3.0

this is plain server
```

This is the test upsptream server:

```js
const express=require("express")
const cors=require("cors")

const app=express()

app.get("/plain",(req,res)=>{
    res.setHeader('Cache-Control', 'max-age=300');

    res.status(200).send("this is plain server")
})

app.listen(4000,()=>{
    console.log("http://localhost:4000")
})
```

**Comment by khangt1k25** [comment]
Oh I figure out the problem. Because of settings for content-type responses. After adding charset-utf 8, it works properly. Thank you. 
< X-Cache-Status: Hit>

**Reply by ms2008** [reply]
Glad it worked out! Actually [Kong's document](https://docs.konghq.com/hub/kong-inc/proxy-cache/configuration/) has made a note of this:

> content_type
> ...
> Note that `*/*` does not match with a content type that contains a parameter (e.g. `*/*` would not match `application/json; charset=utf-8`). If the response content type does not match the configuration, a `Bypass` cache status is returned.
